resources/js/src/app/components/itemList/filter/FindologicItemFilter.spec.ts

```diff
@@ -0,0 +1,351 @@
+
+describe('FindologicItemFilter', () => {
+
+  let store: Store<State>;
+
+  beforeEach(() => {
+    store = new Vuex.Store({});
+  });
+
+  it('Shows the facet name', () => {
+    const facet: Facet = {
+      cssClass: '',
+      findologicFilterType: 'select',
+      id: 'test',
+      isMain: false,
+      itemCount: 3,
+      name: 'Facet name',
+      noAvailableFiltersText: '',
+      select: 'multiple',
+      type: '',
+      values: [
+        {
+          count: 9,
+          id: '20',
+          image: '',
+          items: [],
+          name: '22220',
+          position: 'item',
+          selected: false
+        },
+        {
+          count: 1,
+          id: '21',
+          image: '',
+          items: [],
+          name: '22221',
+          position: 'item',
+          selected: false
+        }
+      ]
+    };
+
+    store.state.itemList = {
+      isLoading: false,
+      selectedFacets: [],
+      facets: [facet]
+    };
+
+    const wrapper = shallowMount(FindologicItemFilter, { propsData: { 'facet': facet, 'filtersPerRow': 3 }, store, localVue });
+
+    expect(wrapper.element.querySelector(':scope > div.h3')!.innerHTML).toBe('Facet name');
+  });
+
+  it('Shows the noAvailableFiltersText when facet has no values', () => {
+    const dropdownFacet: Facet = {
+      cssClass: '',
+      findologicFilterType: 'select',
+      id: 'test',
+      isMain: false,
+      itemCount: 0,
+      name: 'Facet name',
+      noAvailableFiltersText: 'No values available for this dropdown',
+      select: 'multiple',
+      type: '',
+      values: []
+    };
+
+    const categoryFacet: Facet = {
+      cssClass: '',
+      findologicFilterType: 'select',
+      id: 'cat',
+      isMain: false,
+      itemCount: 0,
+      name: 'Category',
+      noAvailableFiltersText: 'No values available for this category dropdown',
+      select: 'single',
+      type: '',
+      values: []
+    };
+
+    const colorFacet: ColorFacet = {
+      cssClass: '',
+      findologicFilterType: 'color',
+      id: 'Color',
+      isMain: false,
+      itemCount: 0,
+      name: 'Color',
+      noAvailableFiltersText: 'No values available for this color filter',
+      select: 'multiselect',
+      type: '',
+      values: []
+    };
+
+    store.state.itemList = {
+      isLoading: false,
+      selectedFacets: [],
+      facets: [dropdownFacet, categoryFacet, colorFacet]
+    };
+
+    const dropdownWrapper = shallowMount(FindologicItemFilter, { propsData: { 'facet': dropdownFacet, 'filtersPerRow': 3 }, store, localVue });
+    const categoryWrapper = shallowMount(FindologicItemFilter, { propsData: { 'facet': categoryFacet, 'filtersPerRow': 3 }, store, localVue });
+    const colorWrapper = shallowMount(FindologicItemFilter, { propsData: { 'facet': colorFacet, 'filtersPerRow': 3 }, store, localVue });
+
+    expect(dropdownWrapper.element.querySelector(':scope > div:last-child p')!.innerHTML).toBe('No values available for this dropdown');
```

Comment:
For easier readability internally we move the `expect()` into a separate method, which is named according to the expectation. For example:

```ts
function expectNoAvailableFiltersText(component) {
  return expect(component.element.querySelector(':scope > div:last-child p')!.innerHTML);
}

// In your test
expectNoAvailableFiltersText(dropdownWrapper).toBe('No values available for this dropdown');
```

Reply:
This may be done for all assertions.

Reply:
Must we really do this? Having to come up with a clear function name for each assertion sounds like a nightmare, to be honest. Perhaps a comment would suffice to clarify some of the more confusing parts?

Reply:
Assertions that happen more than once should definitely have this. If there is a specific case where a div may only exist once on an element, it is probably not required, but still may improve readability of the test, as you immediately know what is being tested without having to see the CSS selector of the element.

Reply:
Please check if the way I've done it works as well. I tried clarifying this by using variables rather than functions as with functions after a couple of assertions it becomes really messy.

Reply:
I'm fine with this for now. Maybe if we change the markup in the future, so we only would need a single selector, we could think about changing this. For now let's leave it like that.